### ISSUE TITLE
TypeScript: Missing semicolon in function type overload

### ISSUE THREAD

**Issue by KingHenne** [prettier/prettier#2003]
Running the [example from the official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) through prettier removes the semicolons at the end of the function type overloads.

**Input**

```ts
function pickCard(x: {suit: string; card: number; }[]): number;
function pickCard(x: number): {suit: string; card: number; };
function pickCard(x): any {
    // Check to see if we're working with an object/array
    // if so, they gave us the deck and we'll pick the card
    if (typeof x == "object") {
        let pickedCard = Math.floor(Math.random() * x.length);
        return pickedCard;
    }
    // Otherwise just let them pick the card
    else if (typeof x == "number") {
        let pickedSuit = Math.floor(x / 13);
        return { suit: suits[pickedSuit], card: x % 13 };
    }
}
```

**Actual Output**

```ts
function pickCard(x: {suit: string; card: number}[]): number
function pickCard(x: number): {suit: string; card: number}
function pickCard(x): any {
  // Check to see if we're working with an object/array
  // if so, they gave us the deck and we'll pick the card
  if (typeof x == 'object') {
    let pickedCard = Math.floor(Math.random() * x.length);
    return pickedCard;
  } else if (typeof x == 'number') {
    // Otherwise just let them pick the card
    let pickedSuit = Math.floor(x / 13);
    return {suit: suits[pickedSuit], card: x % 13};
  }
}
```

**Expected Output**

```ts
function pickCard(x: {suit: string; card: number}[]): number;
function pickCard(x: number): {suit: string; card: number};
function pickCard(x): any {
  // Check to see if we're working with an object/array
  // if so, they gave us the deck and we'll pick the card
  if (typeof x == 'object') {
    let pickedCard = Math.floor(Math.random() * x.length);
    return pickedCard;
  } else if (typeof x == 'number') {
    // Otherwise just let them pick the card
    let pickedSuit = Math.floor(x / 13);
    return {suit: suits[pickedSuit], card: x % 13};
  }
}
```

([interactive example](https://prettier.github.io/prettier/#%7B%22content%22%3A%22function%20pickCard(x%3A%20%7Bsuit%3A%20string%3B%20card%3A%20number%3B%20%7D%5B%5D)%3A%20number%3B%5Cnfunction%20pickCard(x%3A%20number)%3A%20%7Bsuit%3A%20string%3B%20card%3A%20number%3B%20%7D%3B%5Cnfunction%20pickCard(x)%3A%20any%20%7B%5Cn%20%20%20%20%2F%2F%20Check%20to%20see%20if%20we're%20working%20with%20an%20object%2Farray%5Cn%20%20%20%20%2F%2F%20if%20so%2C%20they%20gave%20us%20the%20deck%20and%20we'll%20pick%20the%20card%5Cn%20%20%20%20if%20(typeof%20x%20%3D%3D%20%5C%22object%5C%22)%20%7B%5Cn%20%20%20%20%20%20%20%20let%20pickedCard%20%3D%20Math.floor(Math.random()%20*%20x.length)%3B%5Cn%20%20%20%20%20%20%20%20return%20pickedCard%3B%5Cn%20%20%20%20%7D%5Cn%20%20%20%20%2F%2F%20Otherwise%20just%20let%20them%20pick%20the%20card%5Cn%20%20%20%20else%20if%20(typeof%20x%20%3D%3D%20%5C%22number%5C%22)%20%7B%5Cn%20%20%20%20%20%20%20%20let%20pickedSuit%20%3D%20Math.floor(x%20%2F%2013)%3B%5Cn%20%20%20%20%20%20%20%20return%20%7B%20suit%3A%20suits%5BpickedSuit%5D%2C%20card%3A%20x%20%25%2013%20%7D%3B%5Cn%20%20%20%20%7D%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D))

**Comment by vjeux** [comment]
Thanks, we fixed this in other places but not this one. For anyone who wants to take a stab at it, the fix is to add `semi` if the body is empty.

**Comment by vjeux** [comment]
I just published 1.4.3 which fixes it. Thanks for reporting it!
